File: dx/src/uipp/windows/OptionalPreActionCommand.cpp
/***********************************************************************/
/* Open Visualization Data Explorer                                    */
/* (C) Copyright IBM Corp. 1989,1999                                   */
/* ALL RIGHTS RESERVED                                                 */
/* This code licensed under the                                        */
/*    "IBM PUBLIC LICENSE - Open Visualization Data Explorer"          */
/***********************************************************************/

#include <dxconfig.h>
#include "defines.h"


#include "OptionalPreActionCommand.h"

OptionalPreActionCommand::OptionalPreActionCommand(const char*   name,
				   CommandScope* scope,
				   bool       active,
				   char*         dialogTitle,
				   char*         dialogQuestion):
	PreActionCommand(name, scope, active, 
			dialogTitle, dialogQuestion)
{
}


bool OptionalPreActionCommand::execute(CommandInterface *ci)
{
    //
    // First post the confirmation dialog.
    //
    if (this->needsConfirmation())
	this->PreActionCommand::execute(ci);
    else
	this->NoUndoCommand::execute(ci);
    return false;
}

File: VisualDX/WinDXui/OptionalPreActionCommand.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace WinDX.UI
{
    abstract class OptionalPreActionCommand : PreActionCommand
    {
        protected OptionalPreActionCommand(String name,
            CommandScope scope, bool active, String dialogTitle,
            String dialogQuestion, Form parent)
            : base(name, scope, active, dialogTitle, dialogQuestion, parent)
        { }

        protected OptionalPreActionCommand(String name,
            CommandScope scope, bool active, String dialogTitle,
            String dialogQuestion)
            : base(
            name, scope, active, dialogTitle, dialogQuestion)
        { }

        protected abstract bool needsConfirmation();

        public override bool execute(CommandInterface ci)
        {
            if (needsConfirmation())
            {
                base.execute(ci);
            }
            else
            {
                base.baseExecute(ci);
            }

            return false;
        }
    }
}
